### PR TITLE
feat(tab): add `--calcite-tab-content-block-padding` to override built-in block-padding

### PR DESCRIPTION
**Pull request by Elijbet**
**Related Issue:** #8413

## Summary
Introduces a CSS custom property `--calcite-tab-content-block-padding` to align with other components that have built-in padding. 

Currently, the Tab component has a built-in `block-padding` that cannot be overridden in a supported way, leading to a bug with scrollbars when tabs are housed inside a panel. 

The introduction of the custom property allows for more flexibility in the component, as users can achieve their desired layout and padding by adding it to their slotted content. (https://github.com/Esri/calcite-design-system/issues/8139#issuecomment-1830397218)

### CODE REVIEW

File: packages/calcite-components/src/components/tab/tab.e2e.ts
Comment (jcfranco) [inline packages/calcite-components/src/components/tab/tab.e2e.ts:40]: I think these tests are better suited as screenshot tests as they are more visual.

File: packages/calcite-components/src/components/tab/tab.e2e.ts
Comment (jcfranco) [inline packages/calcite-components/src/components/tab/tab.e2e.ts:49]: This test is misleading because the CSS prop is being set both at the root and component-levels (because of the reused test HTML). Sidebar: [block tests](https://github.com/Esri/calcite-design-system/blob/main/packages/calcite-components/src/components/block/block.e2e.ts#L289-L292) share the same issue. cc @macandcheese

File: packages/calcite-components/src/components/tab/tab.scss
Comment (jcfranco) [inline packages/calcite-components/src/components/tab/tab.scss:6]: CSS prop name and doc does not match the component they belong to.

**Review by jcfranco — CHANGES_REQUESTED** [review]

File: packages/calcite-components/src/components/tab/tab.scss
Comment (Elijbet) [reply packages/calcite-components/src/components/tab/tab.scss:6]: I meant the 'block' here as y-padding. Is there a better way to name it?

Comment (ashetland) [reply packages/calcite-components/src/components/tab/tab.scss:6]: Should the be `--calcite-tab-content-padding`? I think that matches other similar var names.

Comment (jcfranco) [reply packages/calcite-components/src/components/tab/tab.scss:6]: All component-specific tokens share the following naming pattern:

`--calcite-<component>-<prop>`.

Interestingly, I couldn't find this in our conventions, so we should add an entry for this.

Comment (macandcheese) [reply packages/calcite-components/src/components/tab/tab.scss:6]: IMO with the name “content-padding” it should apply a full padding rule if overridden, not just block padding.

File: packages/calcite-components/src/components/tab/tab.e2e.ts
Comment (macandcheese) [reply packages/calcite-components/src/components/tab/tab.e2e.ts:49]: Could you open an issue for that - we can update Block to match the tests here.

File: packages/calcite-components/src/components/tab/tab.scss
Comment (jcfranco) [reply packages/calcite-components/src/components/tab/tab.scss:6]: With the ongoing component-token work, should this be `--calcite-tab-space`, `--calcite-tab-content-space`? cc @alisonailea

File: packages/calcite-components/src/components/tab/tab.e2e.ts
Comment (Elijbet) [reply packages/calcite-components/src/components/tab/tab.e2e.ts:49]: I'll open it right now.

Comment (Elijbet) [reply packages/calcite-components/src/components/tab/tab.e2e.ts:49]: https://github.com/Esri/calcite-design-system/issues/8630

File: packages/calcite-components/src/components/tab/tab.scss
Comment (Elijbet) [reply packages/calcite-components/src/components/tab/tab.scss:6]: Does this not have the same issue as above, where tab-space or tab-content-space would be expected to override full padding around, not just y-padding?

Comment (alisonailea) [reply packages/calcite-components/src/components/tab/tab.scss:6]: @Elijbet if it is only applied to the content element inside tab then the name should be `--calcite-tab-content-space`

Comment (macandcheese) [reply packages/calcite-components/src/components/tab/tab.scss]: This seems like a different approach than other components? IMO it isn’t descriptive for a component level property IMO. Is the “space” referencing space between the tab nav and the tab etc? If not -  Should we change other components that use padding to match this?

Comment (macandcheese) [reply packages/calcite-components/src/components/tab/tab.scss]: > It's the spacing around the tab content, right?
Yeah definitely; just meant it’s not a self explanatory name like “padding”.

Comment (eriklharper) [reply packages/calcite-components/src/components/tab/tab.scss]: > All component-specific tokens share the following naming pattern:
> 
> `--calcite-<component>-<prop>`.
> 
> Interestingly, I couldn't find this in our conventions, so we should add an entry for this.

I was thinking about this the other day and using the Token Naming Schema as a guide, you could expand the template to something like:

`--calcite-<component>-<stencil-prop?>-<css-property?>-<modifier?>`

Where the `?` denote optional, so you could end up with a var like `--calcite-tile-background-color` which leaves out the stencil prop since that isn't applicable, but you could also have a var called `--calcite-tile-heading-text-color` and `--calcite-tile-heading-text-color-hover` which use more of those elements.

Comment (alisonailea) [reply packages/calcite-components/src/components/tab/tab.scss]: Your right I think I need to be more explicit. It is preferable that we create a single `-space` token from which margin and padding are calculated and applied as `-internal-` variables. This should be set on the `:host`. If you are creating a variable that only applied to padding properties then it should end with `-padding`.

Comment (macandcheese) [reply packages/calcite-components/src/components/tab/tab.scss]: For the tile example, it’s also possible the hover color could be set by the user on a pseudo class right? Only internal or inaccessible containers would need to be exposed? Close buttons, etc.

Comment (jcfranco) [reply packages/calcite-components/src/components/tab/tab.scss]: > It is preferable that we create a single -space token from which margin and padding are calculated and applied as -internal- variables. This should be set on the :host. If you are creating a variable that only applied to padding properties then it should end with -padding.

We should decide on one or the other or set some guidelines to deal with the following scenario:

1. component adds `--calcite-component-content-space` (used internally for both margin/padding)
2. [🇫🇷2 hours later🇫🇷] component needs to support different props for margin and padding, so it adds `--calcite-component-content-space-padding` and `--calcite-component-content-space-margin`

Does `--calcite-component-content-space` need to be deprecated? If so, it means we need to support the 3 CSS props until the next breaking change.

LMK if I misread your suggestion.

Comment (jcfranco) [reply packages/calcite-components/src/components/tab/tab.scss]: Regardless, as @Elijbet stated, let's move forward w/ `--calcite-tab-content-padding` for consistency and we can revisit ☝️ soon.

File: packages/calcite-components/src/components/tab/tab.scss
Comment (geospatialem) [inline packages/calcite-components/src/components/tab/tab.scss:6]: Should the css prop description fit the title a bit more? 

Maybe something like:

```suggestion
 * @prop --calcite-tab-content-space: Specifies the spacing of the component's content in the  `"default"` slot.
```

Comment (macandcheese) [reply packages/calcite-components/src/components/tab/tab.scss:6]: Shouldn’t this remain “padding”? Rest of the suggestion looks good.

Comment (geospatialem) [reply packages/calcite-components/src/components/tab/tab.scss:6]: Good catch 🎣 - updated the suggestion!

File: packages/calcite-components/src/components/tab/tab.scss
Comment (jcfranco) [reply packages/calcite-components/src/components/tab/tab.scss]: Actually, this might have to be `--calcite-tab-content-block-padding` (semi full circle, eh? 😅) since the content padding isn't equal and I don't think we can use a single CSS prop for block and inline padding if their values aren't equal.

Comment (jcfranco) [reply packages/calcite-components/src/components/tab/tab.scss]: Got @alisonailea's blessing on ☝️ (via Teams):

> yes. In this case `--calcite-tab-content-block-padding` is probably best

**Review by jcfranco — APPROVED** [review]

**Pull request merged**
